nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureDataLakeStorage.java

```diff
@@ -0,0 +1,243 @@
+            TRACKING_TIME_WINDOW,
+            INITIAL_LISTING_TARGET));
+
+    private volatile Pattern filePattern;
+    private volatile Pattern pathPattern;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        String fileFilter = context.getProperty(FILE_FILTER).getValue();
+        filePattern = fileFilter != null ? Pattern.compile(fileFilter) : null;
+
+        String pathFilter = context.getProperty(PATH_FILTER).getValue();
+        pathPattern = pathFilter != null ? Pattern.compile(pathFilter) : null;
+    }
+
+    @OnStopped
+    public void onStopped() {
+        filePattern = null;
+        pathPattern = null;
+    }
+
+    @Override
+    protected void customValidate(ValidationContext context, Collection<ValidationResult> results) {
+        if (context.getProperty(PATH_FILTER).isSet() && !context.getProperty(RECURSE_SUBDIRECTORIES).asBoolean()) {
+            results.add(new ValidationResult.Builder()
+                    .subject(PATH_FILTER.getDisplayName())
+                    .valid(false)
+                    .explanation(String.format("'%s' cannot be set when '%s' is false", PATH_FILTER.getDisplayName(), RECURSE_SUBDIRECTORIES.getDisplayName()))
+                    .build());
+        }
+    }
+
+    @Override
+    protected RecordSchema getRecordSchema() {
+        return ADLSFileInfo.getRecordSchema();
+    }
+
+    @Override
+    protected Scope getStateScope(PropertyContext context) {
+        return Scope.CLUSTER;
+    }
+
+    @Override
+    protected String getDefaultTimePrecision() {
+        return PRECISION_SECONDS.getValue();
+    }
+
+    @Override
+    protected boolean isListingResetNecessary(PropertyDescriptor property) {
+        return ADLS_CREDENTIALS_SERVICE.equals(property)
+                || FILESYSTEM.equals(property)
+                || DIRECTORY.equals(property);
+    }
+
+    @Override
+    protected String getPath(ProcessContext context) {
+        String directory = context.getProperty(DIRECTORY).evaluateAttributeExpressions().getValue();
+        return directory != null ? directory : ".";
+    }
+
+    @Override
+    protected List<ADLSFileInfo> performListing(ProcessContext context, Long minTimestamp) throws IOException {
+        try {
+            String fileSystem = context.getProperty(FILESYSTEM).evaluateAttributeExpressions().getValue();
+            String baseDirectory = context.getProperty(DIRECTORY).evaluateAttributeExpressions().getValue();
+            boolean recurseSubdirectories = context.getProperty(RECURSE_SUBDIRECTORIES).asBoolean();
+
+            if (StringUtils.isBlank(fileSystem)) {
+                throw new ProcessException(FILESYSTEM.getDisplayName() + " property evaluated to empty string. " +
+                        FILESYSTEM.getDisplayName() + " must be specified as a non-empty string.");
+            }
```

Comment:
This makes me wonder, should the `FILESYSTEM` property be using `NON_BLANK_VALIDATOR` instead of `NON_EMPTY_VALIDATOR`?

Reply:
Indeed, it should. Will change it. Thanks

Reply:
Changed the validator.